docs/source/getting-started/what-is-jupyterhub.md

```diff
@@ -0,0 +1,465 @@
+
+The role of the spawner is to start this server - basically, running
+the command `jupyter notebook`.  Actually it doesn't run that, it runs
+`jupyterhub-singleuser` which first communicates with the hub to say
+"I'm alive" before running a completely normal Jupyter server.  The
+single-user server can be JupyterLab or classic notebooks.  By this
+point, the hub is almost completely out of the picture (the web
+traffic is going through proxy unchanged).  Also by this time, the
+spawner has already decided the environment which this single-user
+server will have and the single-user server has to deal with that.
+
+The spawner starts the server using `jupyterhub-singleuser` with some
+environment variables like `JUPYTERHUB_API_TOKEN` and
+`JUPYTERHUB_BASE_URL` which tell the single-user server how to connect
+back to the hub in order to say that it's ready.
+
+The single-user server options are **JupyterLab** and **classic
+Jupyter Notebook**.  Really, there isn't that much difference between
+them, they run through the same backend server process and the web
+frontend is an option when it is starting.  The spawner can choose the
+command line when it starts the single-user server.  Extensions are a
+property of the single-user server (in two parts: there can be a part
+that runs in the Python server process, and parts that run in
+javascript in lab or notebook).
+
+If one wants to install software for users, it is not a matter of
+"installing it for JupyerHub" - it's a matter of installing it for the
+single-user server, which might be the same environment as the hub,
+but not necessarily.  (Actually, see below - it's a matter of the
+kernels!)
+
+After the single-user notebook server is started, any errors are only
+an issue of the single-user notebook server.  Sometimes, it seems like
+the spawner is failing, but really the spawner is working but the
+single-user notebook server dies right away (in this case, you need to
+find the problem with the single-user server and adjust the spawner to
+start it correctly or fix the environment).  This can happen, for
+example, if the spawner doesn't set an environment variable or doesn't
+provide storage.
+
+The single-user server's logs are handled by the spawner, so if you
+notice problems at this phase you need to check your spawner for
+instructions for accessing the single-user logs.  For example, the
+LocalProcessSpawner logs are just outputted to the same JupyterHub
+output logs (TODO is this correct?), the SystemdSpawner logs are
+written to the Systemd journal, Docker and Kubernetes logs are written
+to Docker and Kubernetes respectively, and batchspawner output goes to
+the normal output places of batch jobs and is an explicit
+configuration option of the spawner.
+
+
+### Notebook
+
+**(Jupyter) Notebook** is the classic interface, where each notebook
+opens in a separate tab.  It is traditionally started by `jupyter
+notebook`.
+
+Does anything need to be said here?
```

Comment:
I think you don't have enough for 2 ###-level sections, maybe just smush them into the single user notebook server.

Reply:
agree - I don't think we need much depth on the interfaces, maybe beyond mentioning that they'll live at different URL prefixes.